components/iam/pkg/server/server.go

```diff
@@ -43,20 +35,18 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
 		return fmt.Errorf("failed to initialize IAM server: %w", err)
 	}
 
-	// All root requests are handled by our router
-	rootHandler, err := registerRootRouter(srv)
+	oidcService, err := oidc.NewServiceWithTestConfig(cfg.OIDCClientsConfigFile)
```

Comment:
I'd prefer to have 2 steps:
1. create service
2. load test config

With DB connection in place, we actually do no longer need the test config.

Reply:
What's the difference in those two steps? The function literally does that - create the service, then add the test config

Reply:
The difference is, that now the service creation in server depends on some config file for tests. 

Anyways, it's temporary. 
So look ahead, I good small step forward would be to introduce a config provider as a dependency to the oidc service. Why? We could start out with backing it by the demo config, but the interesting part is to get to the configs from DB. And in tests we'd somehow mock this provider to not use DB.

Reply:
> depends on some config file for tests.

I'm not sure I follow. Do you have an example?

Reply:
> And in tests we'd somehow mock this provider to not use DB.

We'd just load the config into the DB, like we do in other tests. We already run MySQL as part of CI (and locally) so this should work normally.